agent/config/runtime_test.go

```diff
@@ -0,0 +1,3562 @@
+		pre, post      func()
+		json, jsontail []string
+		hcl, hcltail   []string
+		privatev4      func() ([]*net.IPAddr, error)
+		publicv6       func() ([]*net.IPAddr, error)
+		patch          func(rt *RuntimeConfig)
+		err            string
+		warns          []string
+		hostname       func() (string, error)
+	}{
+		// ------------------------------------------------------------
+		// cmd line flags
+		//
+
+		{
+			desc: "-advertise",
+			flags: []string{
+				`-advertise=1.2.3.4`,
+				`-data-dir=` + dataDir,
+			},
+			patch: func(rt *RuntimeConfig) {
+				rt.AdvertiseAddrLAN = ipAddr("1.2.3.4")
+				rt.AdvertiseAddrWAN = ipAddr("1.2.3.4")
+				rt.RPCAdvertiseAddr = tcpAddr("1.2.3.4:8300")
+				rt.SerfAdvertiseAddrLAN = tcpAddr("1.2.3.4:8301")
+				rt.SerfAdvertiseAddrWAN = tcpAddr("1.2.3.4:8302")
+				rt.TaggedAddresses = map[string]string{
+					"lan": "1.2.3.4",
+					"wan": "1.2.3.4",
+				}
+				rt.DataDir = dataDir
+			},
+		},
+		{
+			desc: "-advertise-wan",
+			flags: []string{
+				`-advertise-wan=1.2.3.4`,
+				`-data-dir=` + dataDir,
+			},
+			patch: func(rt *RuntimeConfig) {
+				rt.AdvertiseAddrWAN = ipAddr("1.2.3.4")
+				rt.SerfAdvertiseAddrWAN = tcpAddr("1.2.3.4:8302")
+				rt.TaggedAddresses = map[string]string{
+					"lan": "10.0.0.1",
+					"wan": "1.2.3.4",
+				}
+				rt.DataDir = dataDir
+			},
+			privatev4: func() ([]*net.IPAddr, error) {
+				return []*net.IPAddr{ipAddr("10.0.0.1")}, nil
+			},
+		},
+		{
+			desc: "-advertise and -advertise-wan",
+			flags: []string{
+				`-advertise=1.2.3.4`,
+				`-advertise-wan=5.6.7.8`,
+				`-data-dir=` + dataDir,
+			},
+			patch: func(rt *RuntimeConfig) {
+				rt.AdvertiseAddrLAN = ipAddr("1.2.3.4")
+				rt.AdvertiseAddrWAN = ipAddr("5.6.7.8")
+				rt.RPCAdvertiseAddr = tcpAddr("1.2.3.4:8300")
+				rt.SerfAdvertiseAddrLAN = tcpAddr("1.2.3.4:8301")
+				rt.SerfAdvertiseAddrWAN = tcpAddr("5.6.7.8:8302")
+				rt.TaggedAddresses = map[string]string{
+					"lan": "1.2.3.4",
+					"wan": "5.6.7.8",
+				}
+				rt.DataDir = dataDir
+			},
+		},
+		{
+			desc: "-bind",
+			flags: []string{
+				`-bind=1.2.3.4`,
+				`-data-dir=` + dataDir,
+			},
+			patch: func(rt *RuntimeConfig) {
+				rt.BindAddr = ipAddr("1.2.3.4")
+				rt.AdvertiseAddrLAN = ipAddr("1.2.3.4")
+				rt.AdvertiseAddrWAN = ipAddr("1.2.3.4")
+				rt.RPCAdvertiseAddr = tcpAddr("1.2.3.4:8300")
+				rt.RPCBindAddr = tcpAddr("1.2.3.4:8300")
+				rt.SerfAdvertiseAddrLAN = tcpAddr("1.2.3.4:8301")
+				rt.SerfAdvertiseAddrWAN = tcpAddr("1.2.3.4:8302")
+				rt.SerfBindAddrLAN = tcpAddr("1.2.3.4:8301")
+				rt.SerfBindAddrWAN = tcpAddr("1.2.3.4:8302")
+				rt.TaggedAddresses = map[string]string{
+					"lan": "1.2.3.4",
+					"wan": "1.2.3.4",
+				}
+				rt.DataDir = dataDir
+			},
+		},
+		{
+			desc: "-bootstrap",
+			flags: []string{
+				`-bootstrap`,
+				`-server`,
+				`-data-dir=` + dataDir,
+			},
+			patch: func(rt *RuntimeConfig) {
+				rt.Bootstrap = true
+				rt.ServerMode = true
+				rt.LeaveOnTerm = false
```

Comment:
In client mode the `LeaveOnTerm` and `SkipLeaveOnInt` values should be the opposite of server mode, so `true` and `false`, respectively. We should add a case that covers that behavior.

Reply:
done.